docs/source/user-guide/expressions/lists-and-arrays.md

```diff
@@ -0,0 +1,182 @@
+# Lists and arrays
+
+Polars has first-class support for two homogeneous container data types: `List` and `Array`.
+Polars supports many operations with the two data types and their APIs overlap, so this section of the user guide has the objective of clarifying when one data type should be chosen in favour of the other.
+
+## Lists vs arrays
+
+### The data type `List`
+
+The data type list is suitable for columns whose values are homogeneous 1D containers of varying lengths.
+
+The dataframe below contains three examples of columns with the data type `List`:
+
+{{code_block('user-guide/expressions/lists', 'list-example', ['List'])}}
+
+```python exec="on" result="text" session="expressions/lists"
+--8<-- "python/user-guide/expressions/lists.py:list-example"
+```
+
+Note that the data type `List` is different from Python's type `list`, where elements can be of any type.
+If you want to store true Python lists in a column, you can do so with the data type `Object` and your column will not have the list manipulation features that we're about to discuss.
+
+### The data type `Array`
+
+The data type `Array` is suitable for columns whose values are homogeneous containers of an arbitrary dimension with a known and fixed shape.
+
+The dataframe below contains two examples of columns with the data type `Array`.
+
+{{code_block('user-guide/expressions/lists', 'array-example', ['Array'])}}
+
+```python exec="on" result="text" session="expressions/lists"
+--8<-- "python/user-guide/expressions/lists.py:array-example"
+```
+
+The example above shows how to use the parameter `schema_overrides` to specify that the columns “bit_flags” and “tic_tac_toe” have the data type `Array`, parametrised by the data type of the elements contained within and by the shape of each array.
```

Comment:
This is not a good example for `schema_overrides` - it's better to just use `schema` here.